File: Semestrul 4/MPP2/Problema8_SC_REST/Problema8SC_Rest/Problema8Networking/src/main/java/problema8/network/jsonprotocol/Response.java
package problema8.network.jsonprotocol;

import problema8.model.AgeGroup;
import problema8.model.Enrollment;
import problema8.network.dto.ChildDTO;
import problema8.network.dto.EnrollmentDTO;
import problema8.network.dto.EventDTO;
import problema8.network.dto.UserDTO;

import java.io.Serializable;

public class Response implements Serializable {
    private ResponseType type;
    private String errorMessage;
    private UserDTO user;
    private ChildDTO child;
    private EnrollmentDTO enrollment;
    private EventDTO event;
    private Object data = null;

    public Response() {
    }
    public ResponseType getType() {
        return type;
    }

    public void setEvent(EventDTO event) {
        this.event = event;
    }
    public EventDTO getEvent() {
        return event;
    }

    public void setType(ResponseType type) {
        this.type = type;
    }
    public String getErrorMessage() {
        return errorMessage;
    }
    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
    public UserDTO getUser() {
        return user;
    }
    public void setUser(UserDTO user) {
        this.user = user;
    }
    public ChildDTO getChild() {
        return child;
    }
    public void setChild(ChildDTO child) {
        this.child = child;
    }

    public Object getData() {
        return data;
    }
    public void setEnrollment(EnrollmentDTO enrollment) {
        this.enrollment = enrollment;
    }
    public EnrollmentDTO getEnrollment() {
        return enrollment;
    }

    public void setData(Object newData) {
        this.data = newData;
    }
}

File: Semestrul 4/MPP2/Problema8_Csharp_ORM/Networking/jsonprotocol/Response.cs
using Networking.dto;

namespace Networking.jsonprotocol;

[Serializable]
public class Response
{
    public ResponseType Type { get; set; }
    public string ErrorMessage { get; set; }
    public UserDTO User { get; set; }
    public ChildDTO Child { get; set; }
    public EnrollmentDTO Enrollment { get; set; }
    public EventDTO Event { get; set; }
    public object Data { get; set; }

    public Response() { }

    public ResponseType GetType()
    {
        return Type;
    }

    public void SetEvent(EventDTO eventObj)
    {
        Event = eventObj;
    }

    public EventDTO GetEvent()
    {
        return Event;
    }

    public void SetType(ResponseType type)
    {
        Type = type;
    }

    public string GetErrorMessage()
    {
        return ErrorMessage;
    }

    public void SetErrorMessage(string errorMessage)
    {
        ErrorMessage = errorMessage;
    }

    public UserDTO GetUser()
    {
        return User;
    }

    public void SetUser(UserDTO user)
    {
        User = user;
    }

    public ChildDTO GetChild()
    {
        return Child;
    }

    public void SetChild(ChildDTO child)
    {
        Child = child;
    }

    public object GetData()
    {
        return Data;
    }

    public void SetEnrollment(EnrollmentDTO enrollment)
    {
        Enrollment = enrollment;
    }

    public EnrollmentDTO GetEnrollment()
    {
        return Enrollment;
    }

    public void SetData(object newData)
    {
        Data = newData;
    }
}